definitions/migrations/1.3.13/atomicmarket.sql

```diff
@@ -0,0 +1,353 @@
+DROP VIEW IF EXISTS atomicmarket_stats_prices_master CASCADE;
+DROP VIEW IF EXISTS atomicmarket_stats_markets_master CASCADE;
+
+DROP TABLE IF EXISTS atomicmarket_stats_markets;
+CREATE TABLE atomicmarket_stats_markets (
+    listing_id bigint not null,
+    price bigint not null,
+    "time" bigint not null,
+    template_id bigint,
+    asset_id bigint,
+	market_contract varchar(12) not null,
+    listing_type text not null,
+    buyer varchar(12) not null,
+    seller varchar(12) not null,
+    maker_marketplace varchar(12) not null,
+    taker_marketplace varchar(12) not null,
+    assets_contract varchar(12) not null,
+    collection_name varchar(12),
+    schema_name varchar(12),
+    symbol varchar(12) not null
+);
+
+
+INSERT INTO atomicmarket_stats_markets (market_contract, listing_type, listing_id, buyer, seller, maker_marketplace, taker_marketplace, assets_contract, collection_name, symbol, price, time,
+    schema_name, template_id, asset_id)
+        SELECT
+            sale.market_contract, 'sale' listing_type, sale.sale_id listing_id,
+            sale.buyer, sale.seller, sale.maker_marketplace, sale.taker_marketplace,
+            sale.assets_contract, sale.collection_name,
+            sale.settlement_symbol symbol, sale.final_price price, sale.updated_at_time "time",
+            CASE WHEN COUNT(*) = 1 THEN MIN(asset.schema_name) END AS schema_name,
+            CASE WHEN COUNT(*) = 1 THEN MIN(asset.template_id) END AS template_id,
+            CASE WHEN COUNT(*) = 1 THEN MIN(asset.asset_id) END AS asset_id
+        FROM atomicmarket_sales sale
+            JOIN atomicassets_offers_assets offer_asset ON sale.offer_id = offer_asset.offer_id AND sale.assets_contract = offer_asset.contract
+            JOIN atomicassets_assets asset ON offer_asset.asset_id = asset.asset_id AND offer_asset.contract = asset.contract
+        WHERE sale.final_price IS NOT NULL AND sale.state = 3
+        GROUP BY sale.market_contract, sale.sale_id
+;
+INSERT INTO atomicmarket_stats_markets (market_contract, listing_type, listing_id, buyer, seller, maker_marketplace, taker_marketplace, assets_contract, collection_name, symbol, price, time,
+    schema_name, template_id, asset_id)
+        SELECT
+            auction.market_contract, 'auction' listing_type, auction.auction_id listing_id,
+            auction.buyer, auction.seller, auction.maker_marketplace, auction.taker_marketplace,
+            auction.assets_contract, auction.collection_name,
+            auction.token_symbol symbol, auction.price, (auction.end_time * 1000) "time",
+            CASE WHEN COUNT(*) = 1 THEN MIN(asset.schema_name) END AS schema_name,
```

Comment:
Just as a question, is this set to NULL if there is more than one in the group?

Reply:
Yes CASE returns NULL when none of the options match